### PR TITLE
fix(deploy): handling jobExecutionError and saving log for SAS9

### PR DESCRIPTION
**Pull request by saadjutt01**
## Issue

no issue to link.
Corresponding to [Adapter PR](https://github.com/sasjs/adapter/pull/657)

## Intent

Should save log in case of `SAS9` deploy having job execution error.
Shouldn't show sas_runner error incorrectly.

## Implementation

Added checks to write correct log into log file, in case of error.

## Checks

- [x] Code is formatted correctly (`npm run lint:fix`).
image
- [ ] Any new functionality has been unit tested.
- [x] All unit tests are passing (`npm test`).
Mocked:
image
Server (without `testing.spec.server`):
image
- [ ] All CI checks are green.
- [ ] JSDoc comments have been added or updated.
- [x] Reviewer is assigned.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -47,8 +47,8 @@
     "access": "public"
   },
   "dependencies": {
-    "@sasjs/adapter": "3.7.4",
-    "@sasjs/core": "4.9.2",
+    "@sasjs/adapter": "file:../adapter/build/sasjs-adapter-5.0.0.tgz",
```
Comment (YuryShkoda) [inline package.json:50]: Adapter shouldn't be installed from a tarball

File: src/commands/deploy/deploy.ts [hunk 2]
```diff
@@ -285,38 +285,58 @@ async function deployToSas9(
...
+        logFolder,
+        `${deployScriptName.replace('.sas', '')}.log`
+      )}`
+    )
+
+    throw new Error(`Deployment failed.`)
+  }
+
+  process.logger?.success(
+    `Deployment completed! Log is available at ${path.join(
```
Comment (YuryShkoda) [inline src/commands/deploy/deploy.ts:330]: `logPath` variable should be created and used on 320 and 330 lines

**Review by YuryShkoda — CHANGES_REQUESTED** [review]

File: src/commands/deploy/deploy.ts (hunk 2 above)
Comment (saadjutt01) [reply src/commands/deploy/deploy.ts]: fixed.

File: package.json (hunk 1 above)
Comment (saadjutt01) [reply package.json:50]: Once, new adapter version is published, I may update this.

Comment (saadjutt01) [reply package.json]: Updated!

**Review by VladislavParhomchik — APPROVED** [review]

**Review by YuryShkoda — APPROVED** [review]

**Pull request merged**